pkgs/development/python-modules/weasel/default.nix

```diff
@@ -0,0 +1,83 @@
+{ lib
+, buildPythonPackage
+, pythonOlder
+, fetchFromGitHub
+, setuptools
+, wheel
+, black
+, cloudpathlib
+, confection
+, isort
+, mypy
+, packaging
+, pre-commit
+, pydantic
+, pytest
+, requests
+, ruff
+, smart-open
+, srsly
+, typer
+, types-requests
+, types-setuptools
+, wasabi
+, pytestCheckHook
+}:
+
+buildPythonPackage rec {
+  pname = "weasel";
+  version = "0.3.3";
+  pyproject = true;
+
+  disabled = pythonOlder "3.6";
+
+  src = fetchFromGitHub {
+    owner = "explosion";
+    repo = "weasel";
+    rev = "refs/tags/v${version}";
+    hash = "sha256-I8Omrez1wfAbCmr9hivqKN2fNgnFQRGm8OP7lb7YClk=";
+  };
+
+  nativeBuildInputs = [
+    setuptools
+    wheel
+  ];
+
+  propagatedBuildInputs = [
+    black
+    cloudpathlib
+    confection
+    isort
+    mypy
+    packaging
+    pre-commit
+    pydantic
+    pytest
+    requests
+    ruff
```

Comment:
The dev and lint dependencies should not be included here. Going to create a PR to fix that.